jane/doc/extensions/_04-modes/syntax.md

```diff
@@ -100,40 +115,126 @@ You can also specify the mode of the function body:
 ```ocaml
 let foo x y @ modes = ..
 let foo x y : ty @ modes = ..
-fun foo x y @ modes -> ..
+fun x y @ modes -> ..
 ```
-We don't support `fun foo x y : ty @ modes -> 42` due to a limitation in the
+We don't support `fun x y : ty @ modes -> 42` due to a limitation in the
 parser.
 
 ## Expressions
 ```ocaml
-(expression : ty @ local)
+(expression : ty @ modes)
 ```
 We don't support `(expression @ modes)` because `@` is already parsed as a binary operator.
+However, you can write `(expression : _ @ modes)` if you do not want to constrain the type.
+
+## Modules
+Support for modules with modes is being worked on and not ready for wide adoption.
+More documentation will come
+as it becomes ready.
+
+# Modalities
+
+Similar to modes, a modality expression is simply a space-delimited list of
+modalities.  As of this writing, every modality is also the name of a mode,
+though it is conceivable we will have modalities other than modes in the future.
+
+```
+modalities ::= modes
+```
+
+Modalities are used to describe the relationship between a container and an
+element in that container; for example, if you have a record field `x` with
+a `portable` modality, then `r.x` is `portable` even if `r` is `nonportable`.
+We say that the `portable` modality applied to the `nonportable` record mode
+produces the `portable` mode of the field. 
+
+Modalities work differently on future axes vs. past axes. On a future axis, the
+modality imposes an upper bound on the mode (thus always lowering that
+mode). Thus applying the `portable` modality to a `nonportable` records yields a
+`portable` field, because `portable < nonportable`. On a past axis, the modality
+imposes a lower bound (thus always raising that mode). Accordingly, a
+`contended` modality applied to an `uncontended` record yields a `contended`
+field, because `uncontended < contended`.
+
+Any axis left out of a modality expression is assumed to be the identity
+modality. (When a modality is the identity, then the mode of a field is the same
+as the mode of the record.) For future axes, this would be the top mode; for
+past axes, this would be the bottom mode. These are the identity modalities:
+
+```ocaml
+local unique once nonportable uncontended unyielding stateless immutable
+```
+
+Note that a legacy mode might or might not be the same as the identity modality.
```

Comment:
While the added text is very informative, it probably belongs to another page dedicated to the *semantics* of modes and modalities. This document was intended as a quick lookup for where and how mode/modalities can show up in syntax.

Reply:
Yes, quite right, thanks. I've added a CR to move it once we have a place to move it to.